product_supplierinfo_for_customer_sale/models/product_customerinfo.py

```diff
@@ -0,0 +1,14 @@
+from odoo import api, fields, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "product.customerinfo"
+
+    product_id = fields.Many2one(
+        "product.product", compute="_compute_product_id", readonly=False
+    )
```

Comment:
```suggestion
    # convert it to computed writable
    product_id = fields.Many2one(
        compute="_compute_product_id", readonly=False, store=True
    )
```

although this can be put in the main module for avoiding incompatibilities.